File: SeekMatch-Client/src/app/shared/models/job-application.ts
import { JobApplicationStatus } from "../enums/enums";
import { ExpressApplication } from "./express-application";
import { JobApplicationStep } from "./job-application-step";

export class JobApplication {

  public id: string | undefined = undefined;
  public appliedAt: string | undefined = undefined;
  public talentId: string = '';
  public talentFullName: string = '';
  public jobOfferId: string = '';
  public jobOfferTitle: string = '';
  public status: JobApplicationStatus = JobApplicationStatus.Submitted;
  public rejectionReason: string = '';
  public isExpress: boolean = false;
  public filePath?: string;
  public interviewPlatform?: string;
  public interviewDate?: string;
  public expressApplication?: ExpressApplication = new ExpressApplication();
  public jobApplicationSteps : JobApplicationStep[] = [];

  constructor(init?:Partial<JobApplication>) {
    Object.assign(this, init);
  }
}

File: SeekMatch-Server/SeekMatch.Core/Entities/Recruiter/JobApplication.cs
using SeekMatch.Core.Enums;

namespace SeekMatch.Core.Entities
{
    public class JobApplication: BaseEntity
    {
        public DateTime AppliedAt { get; set; }
        public string? TalentId { get; set; }
        public Talent? Talent { get; set; }
        public required string JobOfferId { get; set; }
        public JobOffer? JobOffer { get; set; }
        public JobApplicationStatus Status { get; set; } = JobApplicationStatus.Submitted;
        public string? RejectionReason { get; set; }
        public bool IsExpress { get; set; } = false;
        public ExpressApplication? ExpressApplication { get; set; }
    }
}
